lightning/src/chain/keysinterface.rs

```diff
@@ -1143,7 +1143,7 @@ impl KeysInterface for KeysManager {
 	fn ecdh(&self, recipient: Recipient, other_key: &PublicKey, tweak: Option<&[u8; 32]>) -> Result<SharedSecret, ()> {
 		let mut node_secret = self.get_node_secret(recipient)?;
 		if let Some(tweak) = tweak {
-			node_secret.mul_assign(tweak).map_err(|_| ())?;
+			node_secret = node_secret.mul_tweak(&Scalar::from_be_bytes(tweak.clone()).unwrap()).map_err(|_| ())?;
```

Comment:
not sure if there's a way to eliminate this clone op

Reply:
`[u8; 32]` is `Copy`, so `Scalar::from_be_bytes(*tweak).unwrap()` should work well. Even better, you could change the function signature to accept `Option<&Scalar>` instead and move `unwrap` to the caller.

Reply:
ah, yes of course.  will do the former for now and look at the cleanup a bit later